vpr/src/noc/noc_storage.cpp

```diff
@@ -56,6 +56,19 @@ const NocLink& NocStorage::get_single_noc_link(NocLinkId id) const {
     return link_storage[id];
 }
 
+NocLinkId  NocStorage::get_single_noc_link_id(NocRouterId src_router, NocRouterId dst_router) const {
+    NocLinkId link_id = NocLinkId::INVALID();
+
+    for (const auto& link : link_storage) {
```

Comment:
Suggest commenting this is O(Nlinks).

Reply:
Could make it faster with an unordered map keyed by by hash<src_router, dst_router>.

Check how much this is called.

Reply:
This function is called only in unit tests, so its performance does not matter that much.

Reply:
Thanks. Please comment the O(N) behaviour and note that it shouldn't be used in cpu-critical loops.